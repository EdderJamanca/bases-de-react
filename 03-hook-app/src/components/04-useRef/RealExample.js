
import React, { useState } from 'react'
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks';

const RealExample = () => {

    const [ shaw, setShaw]=useState(false);

    return (
        <div>
            <h1>RealExample</h1>
            { shaw && <MultipleCustomHooks />}

            <button 
            onClick={ ()=>{
                setShaw(!shaw)
            }}
            className="btn btn-primary mt-5">
              show/ hide
            </button>

        </div>
    )
}

export default RealExample
