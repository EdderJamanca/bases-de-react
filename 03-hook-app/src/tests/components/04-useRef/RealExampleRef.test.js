import { shallow } from 'enzyme';
import React from 'react';
import RealExample from '../../../components/04-useRef/RealExample';

describe('Prueba en <RealExampleRef />',()=>{

    const wrapper =shallow(<RealExample />);

    test('debe mostrarse correctamente',()=>{

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);

    });

    test('debe de mostrar el componente <MultipleCustomHooks />',()=>{

        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);

    })

})