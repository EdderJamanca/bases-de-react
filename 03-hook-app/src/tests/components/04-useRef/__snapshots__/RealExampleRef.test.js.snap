// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <RealExampleRef /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    RealExample
  </h1>
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    show/ hide
  </button>
</div>
`;
